src/test/java/io/managed/services/test/cluster/DataPlaneClusterTest.java

```diff
@@ -0,0 +1,359 @@
+                break;
+            default:
+                throw new Exception("Unsupported managed kafka type");
+        }
+        log.info("upper limit for instance type '{}' is '{}' instances.", mkType, upperLimitPerManagedKafkaType);
+
+        int currentInstancesCount = FleetshardUtils.getCountOfExistingGivenManagedKafkaCRType(oc, ManagedKafkaType.valueOf(mkType));
+        log.info("currently there are '{}' instances of given type", currentInstancesCount);
+
+        // if there are already max number of instances skip test
+        if (currentInstancesCount > upperLimitPerManagedKafkaType) {
+            log.warn("currently too many instances ({}/{}), of type '{}'",
+                currentInstancesCount,
+                upperLimitPerManagedKafkaType,
+                mkType);
+            throw new SkipException("Too many existing instances, which would cause quota to be breached");
+        }
+
+        // create kafka instance of expected type
+        KafkaRequestPayload payload = new KafkaRequestPayload()
+            .name(KAFKA_INSTANCE_NAME)
+            .cloudProvider("aws")
+            .region("us-east-1")
+            .plan(String.format("%s.x1", mkType));
+
+        log.info("attempt to create kafka instance");
+        KafkaRequest kafkaRequest = null;
+        try {
+            kafkaRequest = kafkaMgmtApi.createKafka(true, payload);
+            log.debug(kafkaRequest);
+            log.info("wait for provisioning of kafka instance with id '{}'", kafkaRequest.getId());
+            KafkaMgmtApiUtils.waitUntilKafkaIsProvisioning(kafkaMgmtApi, kafkaRequest.getId());
+        } catch (ApiGenericException e) {
+            // some users may not be able to create some types of instances, e.g., user with quota will not be able to create dev. instance
+            log.warn(e);
+            JSONObject jsonResponse = new JSONObject(e.getResponseBody());
+            if (KAFKAS_MGMT_21_CODE.equals(jsonResponse.get("code")))
+                throw new SkipException(String.format("user %s has no quota to create instance of type %s", Environment.PRIMARY_USERNAME, mkType));
+            else
+                throw  e;
+        } finally {
+            // cleanup of kafka instance
+            log.info("clean kafka instance with name '{}'", KAFKA_INSTANCE_NAME);
+            try {
+                KafkaMgmtApiUtils.deleteKafkaByNameIfExists(kafkaMgmtApi, KAFKA_INSTANCE_NAME);
+                log.info("wait until kafka is deleted");
+                KafkaMgmtApiUtils.waitUntilKafkaIsDeleted(kafkaMgmtApi, kafkaRequest.getId());
+            } catch (Exception e) {
+                log.error("error while cleaning kafka instance: %s", e);
+            }
+        }
+    }
+
+    @Test()
+    @SneakyThrows
+    public void testStandardKafkaNodeAutoscaling() {
+
+        // make sure dummy instances are cleaned.
+        KafkaMgmtApiUtils.deleteSearchedKafkaInstancesByOwner(kafkaMgmtApi, "jano", Environment.PRIMARY_USERNAME);
```

Comment:
What are these dummy instances?

Reply:
if there is capacity for doing so, we are spawning more instances (to avoid situation when there are some instances deleted, while waiting for autoscaling of node)

Reply:
I see, thanks.